homeassistant/components/greeneye_monitor.py

```diff
@@ -0,0 +1,169 @@
+"""
+Support for monitoring a GreenEye Monitor energy monitor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/greeneye_monitor/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TEMPERATURE_UNIT,
+    EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+
+REQUIREMENTS = ['greeneye_monitor==0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CHANNELS = 'channels'
+CONF_COUNTED_QUANTITY = 'counted_quantity'
+CONF_COUNTED_QUANTITY_PER_PULSE = 'counted_quantity_per_pulse'
+CONF_MONITOR_SERIAL_NUMBER = 'monitor'
+CONF_MONITORS = 'monitors'
+CONF_NET_METERING = 'net_metering'
+CONF_NUMBER = 'number'
+CONF_PULSE_COUNTERS = 'pulse_counters'
+CONF_SERIAL_NUMBER = 'serial_number'
+CONF_SENSORS = 'sensors'
+CONF_SENSOR_TYPE = 'sensor_type'
+CONF_TEMPERATURE_SENSORS = 'temperature_sensors'
+CONF_TIME_UNIT = 'time_unit'
+
+DATA_GREENEYE_MONITOR = 'greeneye_monitor'
+DOMAIN = 'greeneye_monitor'
+
+SENSOR_TYPE_CURRENT = 'current_sensor'
+SENSOR_TYPE_PULSE_COUNTER = 'pulse_counter'
+SENSOR_TYPE_TEMPERATURE = 'temperature_sensor'
+
+TEMPERATURE_UNIT_CELSIUS = 'C'
+
+TIME_UNIT_SECOND = 's'
+TIME_UNIT_MINUTE = 'min'
+TIME_UNIT_HOUR = 'h'
+
+TEMPERATURE_SENSOR_SCHEMA = vol.Schema({
+    vol.Required(CONF_NUMBER): vol.Range(1, 8),
+    vol.Required(CONF_NAME): cv.string,
+})
+
+TEMPERATURE_SENSORS_SCHEMA = vol.Schema({
+    vol.Required(CONF_TEMPERATURE_UNIT): cv.temperature_unit,
+    vol.Required(CONF_SENSORS): cv.ensure_list(TEMPERATURE_SENSOR_SCHEMA),
```

Comment:
Does this work both when passing a dict within a list and just a dict?

We normally do:
```py
vol.Required(CONF_SENSORS): vol.All(
    cv.ensure_list(), [TEMPERATURE_SENSOR_SCHEMA]),
```

Reply:
Good catch. Fixed.